src/wpantund/NCPInstanceBase-Addresses.cpp

```diff
@@ -918,38 +918,80 @@ NCPInstanceBase::find_prefix_entry(const IPv6Prefix &prefix, const OnMeshPrefixE
 	return iter;
 }
 
+std::multimap<NCPInstanceBase::IPv6Prefix, NCPInstanceBase::OnMeshPrefixEntry>::iterator
+NCPInstanceBase::find_domain_prefix_entry(const IPv6Prefix &prefix, const OnMeshPrefixEntry &entry)
+{
+	std::multimap<IPv6Prefix, OnMeshPrefixEntry>::iterator iter;
+
+	iter = mOnMeshPrefixes.begin();
+
+	for (; iter != mOnMeshPrefixes.end(); ++iter)
+	{
+		if (((uint16_t)((*iter).second.get_flags()) & OnMeshPrefixEntry::kFlagExtDP) == OnMeshPrefixEntry::kFlagExtDP)
+		{
+			break;
+		}
+	}
+
+	return iter;
+}
+
+
 void
 NCPInstanceBase::on_mesh_prefix_was_added(Origin origin, const struct in6_addr &prefix_address, uint8_t prefix_len,
 	uint16_t flags, bool stable, uint16_t rloc16, CallbackWithStatus cb)
 {
 	IPv6Prefix prefix(prefix_address, prefix_len);
 	OnMeshPrefixEntry entry(origin, flags, stable, rloc16);
 	std::multimap<IPv6Prefix, OnMeshPrefixEntry>::iterator iter;
+	bool add_new_domain_entry = false;
```

Comment:
Can we rename this as `add_entry` (since it is not just used for `domain_entry`)?